applications/vanilla/controllers/class.discussionscontroller.php

```diff
@@ -722,40 +722,48 @@ public function tagged() {
         $Tag = stringEndsWith($Tag, '.rss', true, true);
         list($Offset, $Limit) = offsetLimit($Page, c('Vanilla.Discussions.PerPage', 30));
 
-        $MultipleTags = strpos($Tag, ',') !== false;
-
         $this->setData('Tag', $Tag, true);
 
         $TagModel = TagModel::instance();
-        $RecordCount = false;
-        if (!$MultipleTags) {
+        //If more than one tag has been passed via GET, pass them as an array to the model.
+        $MultipleTags = strpos($Tag, ',') !== false;
+        if ($MultipleTags) {
+            $Tags = $TagModel->getWhere(array('Name' => explode(',', $Tag)))->resultArray();
+        } else {
             $Tags = $TagModel->getWhere(array('Name' => $Tag))->resultArray();
+        }
 
-            if (count($Tags) == 0) {
-                throw notFoundException('Page');
-            }
+        if (count($Tags) == 0) {
+            throw notFoundException('Page');
+        }
 
-            if (count($Tags) > 1) {
-                foreach ($Tags as $TagRow) {
-                    if ($TagRow['CategoryID'] == val('CategoryID', $Category)) {
-                        break;
-                    }
+        // Get the number of discussions and the page title by either looping
+        // through all the tags or getting it from the result from the TagModel.
+        $RecordCount = false;
```

Comment:
Don't write new code with old standards. You can just update all the variable to lower cases.